implementations/rust/ockam/ockam_api/src/ui/terminal/mod.rs

```diff
@@ -350,62 +350,33 @@ impl<W: TerminalWriter + Debug> Terminal<W, ToStdOut> {
     }
 
     pub fn write_line(mut self) -> Result<()> {
-        // Check that there is at least one output format defined
-        if self.mode.output.plain.is_none()
-            && self.mode.output.machine.is_none()
-            && self.mode.output.json.is_none()
+        let msg = match self
+            .mode
+            .output
+            .get_message(&self.output_format, self.is_tty())?
         {
-            return Err(miette!("At least one output format must be defined"))?;
-        }
-
-        let plain = self.mode.output.plain.clone();
-        let machine = self.mode.output.machine.clone();
-        let json = self.mode.output.json.clone();
-        let (pretty, jq_query) = match self.output_format.clone() {
-            OutputFormat::Json { pretty, jq_query } => (pretty, jq_query),
-            _ => (false, None),
+            Some(msg) => msg,
+            None => return Ok(()),
         };
 
-        let msg = match self.output_format {
-            OutputFormat::Plain => {
-                // If interactive, use the following priority: Plain -> Machine -> JSON
-                if self.stdout.is_tty() {
-                    match (plain, machine, json) {
-                        (Some(plain), _, _) => plain,
-                        (None, Some(machine), _) => machine,
-                        (None, None, Some(json)) => {
-                            self.process_json_output(json, pretty, jq_query.as_ref())?
-                        }
-                        _ => unreachable!(),
-                    }
-                }
-                // If not interactive, use the following priority: Machine -> JSON -> Plain
-                else {
-                    match (machine, json, plain) {
-                        (Some(machine), _, _) => machine,
-                        (None, Some(json), _) => {
-                            self.process_json_output(json, pretty, jq_query.as_ref())?
-                        }
-                        (None, None, Some(plain)) => plain,
-                        _ => unreachable!(),
-                    }
-                }
+        // Log the message in the given format and the plain message if present
+        match (&msg, self.mode.output.plain.as_ref()) {
+            // msg == plain, so we log it once
+            (OutputMessage::Plain(msg), Some(_)) => {
```

Comment:
Small thing: here we can just go `(OutputMessage::Plain(_), Some(_))` and not even do a message comparison.